### PR TITLE
progressive scan

### PR DESCRIPTION
**Pull request by etaty**

### CODE REVIEW

**Comment by dwhjames** [comment]
Looks good. Couple of points.

1. I think `scanOnce` and `scanProgressively` can be unified, such that the former is defined in terms of the latter.
2. `scanOnce` and other methods have code to capture DynamoDB capacity consuming when debug logging is turned on.

**Comment by etaty** [comment]
I think you can merge now

File: src/main/scala/dynamodb/mapper.scala [hunk 1]
```diff
@@ -597,24 +597,17 @@ trait AmazonDynamoDBScalaMapper {
 
     val builder = Seq.newBuilder[T]
 
-    def local(lastKey: Option[DynamoDBKey] = None): Future[Unit] =
-      client.scan(
-        scanRequest.withExclusiveStartKey(lastKey.orNull)
```
Comment (dwhjames) [inline src/main/scala/dynamodb/mapper.scala]: now that you are using `scanProgressively` the construction of `scanRequest` above is dead.

File: src/main/scala/dynamodb/mapper.scala [hunk 2]
```diff
@@ -597,24 +597,17 @@ trait AmazonDynamoDBScalaMapper {
...
-        Option { result.getLastEvaluatedKey } match {
-          case None   => Future.successful(())
-          case optKey => local(optKey)
-        }
+    def local(lastKey: Option[DynamoDBKey] = None): Future[Seq[T]] =
+      scanProgressively(scanFilter, lastEvaluatedKey = lastKey).flatMap {
+        case (key, result) =>
+          builder ++= result
+          key match {
+            case None   => Future.successful(builder.result)
```
Comment (dwhjames) [inline src/main/scala/dynamodb/mapper.scala]: minor: should be `.result()`

**Comment by etaty** [comment]
Oh thanks, should be good now

**Comment by dwhjames** [comment]
:+1:

How urgently did you want to see this published in a new release?

**Comment by etaty** [comment]
It's not urgent, I manually build the mapper with the added scan

**Comment by dwhjames** [comment]
Ok, I’ll endeavor to cut a new release this weekend.

**Pull request merged**